lib/github_changelog_generator.rb

```diff
@@ -24,24 +24,28 @@ module GitHubChangelogGenerator
   class ChangelogGenerator
     # Class, responsible for whole changelog generation cycle
     # @return initialised instance of ChangelogGenerator
-    def initialize
-      @options = Parser.parse_options
-      @generator = Generator.new @options
+    def initialize(params = ARGV)
+      @options   = Parser.parse_options(params)
+      @generator = Generator.new(@options)
     end
 
     # The entry point of this script to generate changelog
     # @raise (ChangelogGeneratorError) Is thrown when one of specified tags was not found in list of tags.
     def run
-      log = @generator.compound_changelog
+      log = generator.compound_changelog
 
       if @options.write_to_file?
         output_filename = @options[:output].to_s
         File.open(output_filename, "wb") { |file| file.write(log) }
         puts "Done!"
-        puts "Generated log placed in #{Dir.pwd}/#{output_filename}"
+        puts "Generated log placed in #{File.absolute_path(output_filename)}"
       else
         puts log
       end
     end
+
+    private
+
+    attr_reader :generator
```

Comment:
If we are going to have private readers, options wants one, too.

Reply:
Either it was private reader of playing around with `instance_variable_set` in the tests. So went with this one to be able to mock the generator. I'll go ahead and use private readers for all the instance variables.